scvi/model/_totalvi.py

```diff
@@ -1249,3 +1260,81 @@ def setup_anndata(
         )
         adata_manager.register_fields(adata, **kwargs)
         cls.register_manager(adata_manager)
+
+    @classmethod
+    def setup_mudata(
+        cls,
+        mdata: MuData,
+        rna_layer: Optional[str] = None,
+        protein_layer: Optional[str] = None,
+        batch_key: Optional[str] = None,
+        size_factor_key: Optional[str] = None,
+        categorical_covariate_keys: Optional[List[str]] = None,
+        continuous_covariate_keys: Optional[List[str]] = None,
+        modalities: Optional[Dict[str, str]] = None,
+        **kwargs,
+    ) -> Optional[AnnData]:
+        setup_method_args = cls._get_setup_method_args(**locals())
+
+        if modalities is None:
+            raise ValueError("Modalities cannot be None.")
+        modalities = modalities.copy()
+        batch_mod = modalities.pop("batch_key", None)
```

Comment:
can the next few lines be automated? Inspect the signature? I like the overall idea though